libs/os/os.cpp

```diff
@@ -0,0 +1,75 @@
+#define HL_NAME(n) os_##n
+#include <hl.h>
+
+#include <clip.h>
+
+HL_PRIM const char * HL_NAME(get_clipboard_text)() {
+    std::string text;
+    clip::get_text(text);
+    return text.c_str();
+}
+
+HL_PRIM bool HL_NAME(set_clipboard_text)(const char * text) {
+    return clip::set_text(text);    
+}
+
+HL_PRIM vbyte * HL_NAME(get_clipboard_image_data)() {
+    clip::image img;
+    if(clip::get_image(img)) {
+        int size = img.spec().width * img.spec().height * (img.spec().bits_per_pixel / 8) * sizeof(char);
+        vbyte * data = (vbyte *)hl_gc_alloc_raw(size);
```

Comment:
I don't know, do you have to use hl_gc_alloc... functions, so that the memory is freed at some point or if we are returning bytes from the function, will it take care of them automatically?

Reply:
This usage is correct. The `vbyte*` (= `hl.Bytes`) block is known to the GC and will be collected when no longer reachable.

Reply:
Ok, thanks. :)